File: SnappetChallenge.Core/Models/LearningObjective.cs
namespace SnappetChallenge.Core.Models
{
    public class LearningObjective
    {
        public string Name { get; set; }
        public string Domain { get; set; }
        public string Subject { get; set; }
        public UserForLearningObjective[] Users { get; set; }
    }
}

File: SnappetChallenge/wwwroot/scripts/models/learningObjective.ts
module SnappetChallenge.Models {
    export class LearningObjective {
        name: string;
        domain: string;
        subject: string;
        averageProgress: number;
        users: UserForLearningObjective[];
    }

    export class UserForLearningObjective {
        userId: number;
        name: string;
        overallProgress: number;
        imageId: number;
    }
}